enigma/src/main/java/cuchaz/enigma/translation/mapping/IdentifierValidation.java

```diff
@@ -23,12 +22,13 @@
 	private IdentifierValidation() {
 	}
 
-	private static final List<String> ILLEGAL_IDENTIFIERS = Arrays.asList(
-			"abstract", "continue", "for", "new", "switch", "assert", "default", "goto", "package", "synchronized",
-			"boolean", "do", "if", "private", "this", "break", "double", "implements", "protected", "throw", "byte",
-			"else", "import", "public", "throws", "case", "enum", "instanceof", "return", "transient", "catch",
-			"extends", "int", "short", "try", "char", "final", "interface", "static", "void", "class", "finally",
-			"long", "strictfp", "volatile", "const", "float", "native", "super", "while", "true", "false", "null", "_"
+	private static final List<String> ILLEGAL_IDENTIFIERS = List.of(
```

Comment:
Could you add "record" to this (any other keywords that got added since java 8, not sure right now if there's more)?

Reply:
'sealed', 'permits' and 'record' is all that come to my mind.

Reply:
apparenly, 'non-sealed' is one as well.

Reply:
No? I recall those are contextual keywords that are only recognized in certain positions.
https://docs.oracle.com/javase/specs/jls/se16/html/jls-3.html#jls-3.9

They are called "restricted identifiers" per https://docs.oracle.com/javase/specs/jls/se16/html/jls-3.html#jls-3.8

Reply:
Ah okay, good to know.

Reply:
Yeah, modern java is tricky as it avoids to break existing code :trollface:

Reply:
Java seems to have its own list, SourceVersion.isKeyword none of these appear there so it should be fine as is 👍

Reply:
Note that even though `record` is not a full keyword, it still can't be used as a type name:
```
public class record {
             ^
  as of release 14, 'record' is a restricted type name and cannot be used for type declarations
```

Reply:
Yeah, a right way would be to make `validateClassName` and `validateMethodName` etc have different implementations.